python/lsst/pipe/tasks/makeCoaddTempExp.py

```diff
@@ -58,9 +58,150 @@ class MakeCoaddTempExpConfig(CoaddBaseTask.ConfigClass):
     )
 
+## \addtogroup LSST_task_documentation
+## \{
+## \page MakeCoaddTempExpTask
+## \ref MakeCoaddTempExpTask_ "MakeCoaddTempExpTask"
+## \copybrief MakeCoaddTempExpTask
+## \}
+
 
 class MakeCoaddTempExpTask(CoaddBaseTask):
-    """Task to produce <coaddName>Coadd_<warpType>Warp images
+    """!Warp and optionally PSF-Match calexps onto an a common projection.
+
+    @anchor MakeCoaddTempExpTask_
+
+    @section pipe_tasks_makeCoaddTempExp_Contents  Contents
+
+     - @ref pipe_tasks_makeCoaddTempExp_Purpose
+     - @ref pipe_tasks_makeCoaddTempExp_Initialize
+     - @ref pipe_tasks_makeCoaddTempExp_IO
+     - @ref pipe_tasks_makeCoaddTempExp_Config
+     - @ref pipe_tasks_makeCoaddTempExp_Debug
+     - @ref pipe_tasks_makeCoaddTempExp_Example
+
+    @section pipe_tasks_makeCoaddTempExp_Purpose  Description
+
+    Warp and optionally PSF-Match calexps onto a common projection, by
+    performing the following operations:
+    - Group calexps by visit/run
+    - For each visit, generate a Warp by calling method @ref makeTempExp.
+      makeTempExp loops over the visit's calexps calling @ref WarpAndPsfMatch
+      on each visit
+
+    The result is a `directWarp` (and/or optionally a `psfMatchedWarp`).
+
+    @section pipe_tasks_makeCoaddTempExp_Initialize  Task Initialization
+
+    @copydoc \_\_init\_\_
+
+    This task has no special keyword arguments.
+
+    @section pipe_tasks_makeCoaddTempExp_IO  Invoking the Task
+
+    This task is primarily designed to be run from the command line.
+
+    The main method is `run`, which takes a single butler data reference for the patch(es)
+    to process.
+
+    @copydoc run
+
+    @copydoc WarpType
+
+    @section pipe_tasks_makeCoaddTempExp_Config  Configuration parameters
+
+    See @ref MakeCoaddTempExpConfig and parameters inherited from
+    @link lsst.pipe.tasks.coaddBase.CoaddBaseConfig CoaddBaseConfig @endlink
+
+    @subsection pipe_tasks_MakeCoaddTempExp_psfMatching Guide to PSF-Matching Configs
+
+    To make `psfMatchedWarps`, select `config.makePsfMatched=True`. The subtask
+    @link lsst.ip.diffim.modelPsfMatch.ModelPsfMatchTask ModelPsfMatchTask @endlink
+    is responsible for the PSF-Matching, and its config is accessed via `config.warpAndPsfMatch.psfMatch`.
+    The optimal configuration depends on aspects of dataset: the pixel scale, average PSF FWHM and
+    dimensions of the PSF kernel. These configs include the matching kernel size, the requested model PSF,
+    padding of the science PSF thumbnail and spatial sampling frequency of the PSF.
+    The following list presents the most common problems that arise from a misconfigured ModelPsfMatchTask
+    and corresponding solutions. All assume the default Alard-Lupton kernel, with configs accessed via
+    ```config.warpAndPsfMatch.psfMatch.kernel['AL']```.
+
+    *Troublshooting PSF-Matching Configuration:*
+    - Matched PSFs look boxy: The matching kernel is too small. _Increase the matching kernel size.
```

Comment:
Added a paragraph giving some guidelines for setting defaultFwhm of the modelPSF (which is science-dependent and users will usually have to change for each camera/observing-run) and a statement that the defaults should be OK after that.